config/config.js

```diff
@@ -0,0 +1,23 @@
+/*
+* config.js configures the port number, mysql connection and bot connection
+*/
+var mysql = require('mysql');
+var irc = require('irc');
+var config = {
+  port: 8000, //port number
+  //create a mysql connection
+  conn: mysql.createConnection({
```

Comment:
This conn is being used across the server. 
More than  one user logging at the same time and sending events would be a bottleneck. Either create conn when you need them or use a connection pool. In your case you create connection at the server start itself as all the modules are imported at boot. Can you change this ?